launcher/minecraft/mod/tasks/LocalModParseTask.cpp

```diff
@@ -401,6 +428,36 @@ bool processZIP(Mod& mod, ProcessingLevel level)
 
         mod.setDetails(details);
         return true;
+    } else if (zip.setCurrentFile("META-INF/nil/mappings.json")) {
+        // nilloader uses the filename of the metadata file for the modid, so we can't know the exact filename
+        // thankfully, there is a good file to use as a canary so we don't look for nil meta all the time
```

Comment:
Turns out it could be possible to at some point find jars without this file in the wild - I've been messing around with making an annotated config library mod for nil, which does not require any mappings as it doesn't touch anything outside NilLoader itself. In this case, the mappings file is not inserted into the jar since it does not exist. Not entirely sure what to do about this.
(This would also be the case for mods for games that do not require remapping, but prism is not going to be launching anything other than minecraft.)